lib/src/lib.rs

```diff
@@ -660,6 +676,10 @@ impl Repo {
         let repo = Repo {
             path: repo_path.to_owned(),
             pub_key: pk,
+            write_stats: Arc::new(Mutex::new(WriteStats {
```

Comment:
The main thread is already joining on `ThreadHandle<T>` where `T` is result of a thread. So most idiomatic would be for writing thread to return the stats and main thread to collect it on `join`.

Though I prefer the one that you decide to implement. :) . If you ask me, you're right about it probably not being significant. You can just add a comment with a link to here and call it a day. I don't think it's important, and you're doing a lot of more significant things. :)